FR #962: Add Correction Handling and Delivery Handling attributes and refactor Invoice Handling (#1501)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/invoice_detail/dataset.md

  * InvoiceDetail MUST include [PurchaseOrderNumber](#datasets.invoicedetail.purchaseordernumber) if the invoice issuer supports customer input of purchase order numbers.
  * InvoiceDetail MUST include [ReferenceInvoiceId](#datasets.invoicedetail.referenceinvoiceid).
* InvoiceDetail MUST conform to [ColumnHandling](#attributes.columnhandling) requirements.
* InvoiceDetail MUST conform to [CorrectionHandling](#attributes.correctionhandling) requirements.
* InvoiceDetail MUST conform to [DatasetCompleteness](#attributes.datasetcompleteness) requirements.
* InvoiceDetail MUST conform to [DeliveryHandling](#attributes.deliveryhandling) requirements.
* InvoiceDetail MUST conform to [NullHandling](#attributes.nullhandling) requirements.
* InvoiceDetail MUST represent all invoice line items with a non-zero BilledCost on any invoice associated with a BillingAccountId.

> * InvoiceDetail MUST include custom columns to represent any monetary metric that appears on an invoice issued to a BillingAccountId when there is no equivalent FOCUS column.

Reviewer: ```suggestion
* InvoiceDetail MUST include [*custom columns*](#glossary:custom-column) to represent any monetary metric that appears on an invoice issued to a BillingAccountId when there is no equivalent [*FOCUS column*](#glossary:FOCUS-column).
```

Both "custom columns" and "FOCUS column" are glossary terms. First occurrence of each in the Requirements section should use glossary link syntax. Matches the pattern in dataset_completeness.md: `[*custom columns*](#glossary:custom-column)`.

  * InvoiceDetail MUST include [ReferenceInvoiceId](#datasets.invoicedetail.referenceinvoiceid).

> * InvoiceDetail MUST conform to [ColumnHandling](#attributes.columnhandling) requirements.

Author: These should be alpha-sorted, right?  Like so?

```
* InvoiceDetail MUST conform to [ColumnHandling](#attributes.columnhandling) requirements.
* InvoiceDetail MUST conform to [CorrectionHandling](#attributes.correctionhandling) requirements.
* InvoiceDetail MUST conform to [DeliveryHandling](#attributes.deliveryhandling) requirements.
* InvoiceDetail MUST conform to [NullHandling](#attributes.nullhandling) requirements.
```

Reviewer 2: I agree, we should revisit/simplify this section in the guidelines and add a **Conformance to Attribute** section with alphabetical ordering.

<img width="1162" height="356" alt="image" src="https://github.com/user-attachments/assets/b2f1583e-b16f-46c4-a03e-df9d8c986364" />

Added [a reminder](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1967/files#r2995156819) to address this guidelines section in PR #1234.

I’ll update the ordering in both datasets.

Reviewer 2: reordered in all 3 dataset.md files

Reviewer 2: @shawnalpay @flanakin should InvoiceDetail conform to [DatasetConfiguration](#attributes.datasetconfiguration) requirements?

Author: Reorder looks good.

And yes -- all the datasets should conform to DatasetConfiguration requirements.

Reviewer 2: It's missing. Should I add it here or is that being addressed in some other PR?

Author: Don't do it here.  I'll stand up a separate PR right now.

Author: Stood up #2113.

* InvoiceDetail MUST conform to [NullHandling](#attributes.nullhandling) requirements.
* InvoiceDetail MUST conform to [CorrectionHandling](#attributes.correctionhandling) requirements.
* InvoiceDetail MUST conform to [DeliveryHandling](#attributes.deliveryhandling) requirements.

> * InvoiceDetail MUST represent all invoice line items with a non-zero BilledCost on any invoice associated with a BillingAccountId.

Author: ```suggestion
* InvoiceDetail MUST represent all invoice line items with a non-zero [BilledCost](#datasets.invoicedetail.billedcost) on any invoice associated with a [BillingAccountId](#datasets.invoicedetail.billingaccountid).
```

Reviewer 2: Reminder:
According to Editorial Guidelines **The first occurrence in a section is linked to the section.**
Both of these are mentioned earlier in this document and even, i.e., included in previous requirements.

Author: Heard.  In that case, the next line (line 78) needs its BillingAccountId link removed.  Once you do that, you can resolve this thread.

Reviewer 2: addressed. resolving

> * InvoiceDetail MUST include custom columns to represent any cost or monetary metric that appears on an invoice issued to a BillingAccountId when there is no equivalent FOCUS column.

Author: @ijurica 
Agree with your suggestion to just make this "monetary".

```suggestion
* InvoiceDetail MUST include custom columns to represent any monetary metric that appears on an invoice issued to a [BillingAccountId](#datasets.invoicedetail.billingaccountid) when there is no equivalent FOCUS column.
```

Reviewer 2: **Reminder:**  
According to the Editorial Guidelines, **“The first occurrence in a section is linked to the section.”**

BillingAccountId is already mentioned and linked earlier in this document.

I’ve added [a note](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/1878#issuecomment-4134920469) in #1878 to review and apply this consistently throughout the spec.

> * InvoiceDetail MUST include a custom column to represent any cost or monetary metric that appears on an invoice issued to a BillingAccountId when there is no equivalent FOCUS column.

Reviewer 2: ```suggestion
* InvoiceDetail MUST include custom columns to represent any cost or monetary metric that appears on an invoice issued to a BillingAccountId when there is no equivalent FOCUS column.
```

> * InvoiceDetail MUST include a custom column (e.g., x_PricingQuantity) to represent any metric that appears on an invoice issued to a BillingAccountId when there is no equivalent FOCUS column.

Reviewer 2: ```suggestion
* InvoiceDetail MUST include a custom column to represent any cost or monetary metric that appears on an invoice issued to a BillingAccountId when there is no equivalent FOCUS column.
```
